src/transaction/transaction.rs

```diff
@@ -336,6 +337,9 @@ impl Transaction {
     /// # });
     /// ```
     pub async fn lock_keys(&self, keys: impl IntoIterator<Item = impl Into<Key>>) -> Result<()> {
+        if self.is_pessimistic {
+            return Err(ErrorKind::InvalidTransactionType.into());
+        }
```

Comment:
This change should be in a separate PR, I think

Reply:
OK